### PR TITLE
RS/YJ/CHECK_PURCHASED_CHW_HHW

### PR DESCRIPTION
**Pull request by yunjoonjung-PNNL**
I'd appreciate it if you reviewed this PR. Thanks!

### CODE REVIEW

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:33]: Fixed.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:41]: Fixed.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py]: Fixed.

**Review by yunjoonjung-PNNL — COMMENTED** [review]
I addressed all the PR comments. I'd appreciate it if you could review again.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (charliepnnl) [inline rct229/ruleset_functions/check_purchased_chw_hhw.py:63]: I have the same comment here as above regarding the RDS indicating that all child loops should be added to the list.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (charliepnnl) [inline rct229/ruleset_functions/check_purchased_chw_hhw.py:80]: I believe hvac_sys["preheat_system"] is not a list. It looks like the RDS needs to be fixed too.

**Review by charliepnnl — APPROVED** [review]
Approved with a few comments that need to be addressed before merging

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:12]: Fixed.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:24]: Fixed.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:10]: @charliepnnl @weilixu Should we also get approval from the RDS team to change the function name? I see ```check_purchased_chw_hhw``` function name in [RDS](https://github.com/pnnl/ruleset-checking-tool/blob/9df71942341b6f6e4c727a3e80ff7254b20f53d5/docs/ruleset_functions/check_purchased_chw_hhw.md#:~:text=Blame-,check_purchased_chw_hhw,-Description%3A%20Check%20if), and am not sure if I can change the name by myself.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:32]: Fixed.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:31]: Great point! Thanks for letting me know about this!

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:36]: Fixed.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:40]: I appreciate for catching this up!

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:60]: Fixed.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:38]: Fixed.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:43]: The result is ```None``` so, I modified the code to ```find_all("child_loops[*].id", cooling_loop) + [cooling_loop["id"]]```.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:63]: Fixed.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:80]: True! Fixed.

File: rct229/ruleset_functions/baseline_systems/baseline_system_util.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/baseline_systems/baseline_system_util.py:145]: Yes! I corrected this.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (yunjoonjung-PNNL) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:58]: Fixed.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (weilixu) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:11]: function name should be OK. We have done similar changes in some other functions. Thanks for checking with us.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (weilixu) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py]: We do not need `getattr_` in this case since [`loop`](https://github.com/open229/ruleset-model-description-schema/blob/master/docs229/ASHRAE229.schema.json#L3797) is a required field under the `external_fluid_loop`.

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (weilixu) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py:60]: @charliepnnl I am not sure if assertion is needed here since `type` is an enum. Does the schema validation at the beginning of evaluation return INVALID_ error if the string does not match to the enum value?

File: rct229/ruleset_functions/check_purchased_chw_hhw.py
Comment (weilixu) [reply rct229/ruleset_functions/check_purchased_chw_hhw.py]: We should be ok with loop :)

**Review by weilixu — APPROVED** [review]
Approved with some comments need to be addressed.

**Pull request merged**
